### PR TITLE
Add Authorize Upgrade Pattern to Frame System

### PR DESCRIPTION
**Pull request by joepetrowski**
Adds the `authorize_upgrade` -> `enact_authorized_upgrade` pattern to `frame-system`. This will be useful for upgrading bridged chains that are under the governance of Polkadot without passing entire runtime Wasm blobs over a bridge.

Notes:

- Changed `enact_authorized_upgrade` to `apply_authorized_upgrade`. Personal opinion, "apply" more accurately expresses what it's doing. Can change back if outvoted.
- Remove `check_version` in favor of two extrinsics, so as to make _checked_ the default.
- Left calls in `parachain-system` and marked as deprecated to prevent breaking the API. They just call into the `frame-system` functions.
- Updated `frame-system` benchmarks to v2 syntax.

### CODE REVIEW

**Comment by joepetrowski** [comment]
bot fmt

**Review by muharem — APPROVED** [review]

**Comment by muharem** [comment]
minor comment, is that we have now different naming for one group of actions within the same pallet, `set_code` and `authorized_upgrade`. may be worth mentions in pallet's docs

File: substrate/frame/system/src/lib.rs [hunk 1]
```diff
@@ -1872,6 +2004,46 @@ impl<T: Config> Pallet<T> {
...
+			// no fee for valid upgrade
+			pays_fee: Pays::No,
+		};
+		Ok(post)
+	}
+
+	/// Return an upgrade authorization, if it exists.
+	pub fn authorized_upgrade() -> Option<CodeUpgradeAuthorization<T>> {
+		AuthorizedUpgrade::<T>::get()
+	}
```
[the remark this answers is not in the document]
Comment (joepetrowski) [reply substrate/frame/system/src/lib.rs:2034]: I'm not sure, will it make it public to other pallets?

[the remark this answers is not in the document]
Comment (liamaharon) [reply substrate/frame/system/src/lib.rs:2034]: Should do

https://github.com/paritytech/polkadot-sdk/blob/0182192d877fc81d3ea88a4386bcb4ae3308ec0c/substrate/frame/support/procedural/src/pallet/expand/storage.rs#L475-L479

[the remark this answers is not in the document]
Comment (joepetrowski) [reply substrate/frame/system/src/lib.rs]: Yup good call, fixed.

**Review by liamaharon — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-6-0/5855/1

**Pull request merged**